backend/tracim_backend/migration/versions/8a9d929b44b9_add_external_id_to_user.py

```diff
@@ -0,0 +1,29 @@
+"""add external_id to user
+
+Revision ID: 8a9d929b44b9
+Revises: f4d030ffd43e
+Create Date: 2023-11-08 16:21:10.532016
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+# revision identifiers, used by Alembic.
+revision = "8a9d929b44b9"
+down_revision = "f4d030ffd43e"
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table("users") as batch_op:
+        batch_op.add_column(sa.Column("external_id", sa.VARCHAR(length=255), nullable=True))
+        batch_op.create_unique_constraint(op.f("uq__users__external_id"), ["external_id"])
```

Comment:
In the issue description (in bonus part) I suggested *(do not force uniqueness for now, but allow to identify a potential problem)*

Are you sure we can have uniqueness + several lines with a NULL value? (for mysql, sqlite and postgresql)? I have a vague memory of some issue with this kind of situation (maybe it's not an issue today)

Reply:
according to your answer we could keep uniqueness

Reply:
I'm not sure that's the issue :/ I know that it is ok for postgres but not the rest.
I suggested to fill the values with the users's actual user_id but we agree that it could conflict with real external_ids . I'm not sure of what to do from now on

Reply:
We should remove uniqueness then ?